provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/endpoint/EndpointController.java

```diff
@@ -1,3 +1,152 @@
+            .totalRecords(root.getRight())
+            .build();
+    return bundler.bundle(
+        BundleContext.of(
+            linkConfig, root.getLeft(), transformer, Endpoint.Entry::new, Endpoint.Bundle::new));
+  }
+
+  private Pair<List<EndpointWrapper>, Integer> search(MultiValueMap<String, String> parameters) {
+    if (parameters.containsKey("name")) {
+      return searchName(parameters);
+    } else if (parameters.containsKey("identifier")) {
+      return searchIdentifier(parameters);
+    } else {
+      return searchOrganization(parameters);
+    }
+  }
+
+  /** Placeholder for Identifier search. */
+  @GetMapping(params = {"identifier"})
+  public Endpoint.Bundle searchByIdentifier(
+      @RequestParam("identifier") String identifier,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @RequestParam(value = "_count", defaultValue = "15") @Min(0) int count) {
+    throw new UnsupportedOperationException();
+  }
+
+  /** Search by Name. */
+  @GetMapping(params = {"name"})
+  public Endpoint.Bundle searchByName(
+      @RequestParam("name") String name,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder().add("name", name).add("page", page).add("_count", count).build(),
+        page,
+        count);
+  }
+
+  /** Placeholder for Organization search. */
+  @GetMapping(params = {"organization"})
+  public Endpoint.Bundle searchByOrganization(
+      @RequestParam("organization") String organization,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @RequestParam(value = "_count", defaultValue = "15") @Min(0) int count) {
+    throw new UnsupportedOperationException();
+  }
+
+  /** Placeholder for search by Identifier. */
+  private Pair<List<EndpointWrapper>, Integer> searchIdentifier(
+      MultiValueMap<String, String> parameters) {
+    return null;
+  }
+
+  /** Logic for search by Name. */
+  private Pair<List<EndpointWrapper>, Integer> searchName(
+      MultiValueMap<String, String> parameters) {
+    String name = parameters.getFirst("name");
+    AddressResponse addressResponse = vlerClient.endpointByAddress(name);
+    int page = pageOf(parameters);
+    int count = countOf(parameters);
+    int fromIndex = Math.min((page - 1) * count, addressResponse.contacts().size());
+    int toIndex = Math.min((fromIndex + count), addressResponse.contacts().size());
+    List<AddressResponse.Contacts> addressResponsePages =
+        addressResponse.contacts().subList(fromIndex, toIndex);
+    List<EndpointWrapper> endpointWrapperPages = new ArrayList<>();
+    for (int i = 0; i < addressResponsePages.size(); i++) {
+      endpointWrapperPages.add(
+          (EndpointWrapper.builder()
+                  .addressResponse(
+                      AddressResponse.builder()
+                          .contacts(singletonList(addressResponsePages.get(i)))
```

Comment:
I made a similar comment on a previous PR, but it is unnecessarily complicated here to create this fake `AddressResponse` that has a singleton list containing `addressResponsePages.get(i)`. It is more straightforward (both here and in the transformer later) to change `EndpointWrapper` to have a field that is just the single `addressResponsePages.get(i)` value instead.